kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/ConfigurationDefinition.java

```diff
@@ -9,7 +9,7 @@
 import io.kroxylicious.proxy.config.BaseConfig;
 
 /**
- * Defines the details of how Contributions interact with configuration for {@link BaseContributor}'s purposes.
+ * Defines the details of how Contributions interact with configuration.
  * <p>
  * configurationRequired allows filter authors to tell Kroxylicious if the filter must be supplied with a config object.
  * If the filter uses configuration to supply optional properties, or it can provide sensible default values for the configuration then it should be set to false.
```

Comment:
I don't understand why we need this. 
* The runtime parses the supplied config file which has a number of filters, each with a mandatory `type` and some optional `config`. 
* The `type` unambiguously identifies an implementation of the `FilterContributor` interface (which to me is really just a factory, and ought to be called `FilterFactory`)
* From the implementation of the FilterContributor we get the config type (lets call it `C`), which allows jackson to convert the untyped `JsonNode` to that config type. The most specific type we can use in that code is `Object`, but it's actually a `C`
* The `FilterContributor` also exposes a `getInstance()` method. So the implementor of FilterContributor (i.e. the filter author) gets a chance to accept or reject (via an exception) a null config object at that point. 

Now we can argue about what's the best signature for `getInstance()`, should be it `(C config, FilterConstructContext context)` or `(FilterConstructContext<C> context)` or something which forces the filter author to down cast, like `(Object config, FilterConstructContext context)`, but in any case I don't see a need for this `ConfigurationDefinition` if all it does it support optional config.

Reply:
For some Context the declaration of optionality was part of a fix for #530

Reply:
With this cardinality change it might make sense for it to now become a `boolean requiresConfiguration()` method on `Contributor`. Before we needed some class to represent the needs of each type of thing in a Contributor that offered N service types.

Reply:
Yes, I think that makes sense. I think you've got two possible models:

1. Try to prevalidate the call to `getInstance()`, which is what this flag lets you do. Benefit: You can provide a clear and consistent error message to the proxy configurer about how they screwed up
2. Cope (better than propagating an NPE) with exceptions from `getInstance`.

They're not mutually exclusive, of course. You probably need 2 in any case.